client/lib/analytics/ad-tracking.js

```diff
@@ -250,87 +250,63 @@ function setupOutbrainGlobal() {
 	api.queue = [];
 }
 
+const loadScript = promisify( loadScriptCallback );
+
 function loadTrackingScripts( callback ) {
 	hasStartedFetchingScripts = true;
 
 	const scripts = [];
 
 	if ( isFacebookEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( FACEBOOK_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( FACEBOOK_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isAdwordsEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( GOOGLE_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( GOOGLE_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isBingEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( BING_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( BING_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isCriteoEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( CRITEO_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( CRITEO_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isQuantcastEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( quantcastAsynchronousTagURL(), onComplete );
-		} );
+		scripts.push( loadScript( quantcastAsynchronousTagURL() ) );
 	}
 
 	if ( isYahooEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( YAHOO_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( YAHOO_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isTwitterEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( TWITTER_TRACKING_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( TWITTER_TRACKING_SCRIPT_URL ) );
 	}
 
 	if ( isLinkedinEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( LINKED_IN_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( LINKED_IN_SCRIPT_URL ) );
 	}
 
 	if ( isMediaWallahEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( MEDIA_WALLAH_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( MEDIA_WALLAH_SCRIPT_URL ) );
 	}
 
 	if ( isQuoraEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( QUORA_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( QUORA_SCRIPT_URL ) );
 	}
 
 	if ( isYandexEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( YANDEX_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( YANDEX_SCRIPT_URL ) );
 	}
 
 	if ( isOutbrainEnabled ) {
-		scripts.push( function( onComplete ) {
-			loadScript( OUTBRAIN_SCRIPT_URL, onComplete );
-		} );
+		scripts.push( loadScript( OUTBRAIN_SCRIPT_URL ) );
 	}
 
-	async.series( scripts, function( error ) {
-		if ( error ) {
-			debug( 'Some scripts failed to load: ', error );
-		} else {
+	Promise.all( scripts ).then(
```

Comment:
`async.series` loads the scripts one by one, waiting for the previous load to succeed before starting the next one. `Promise.all` loads them all in parallel.

Does the difference matter? I don't know. If the scripts depend on each other, then the order in which they are loaded is important.

This code would be much closer to the original `async.series`:
```js
try {
  for ( const script of scripts) {
    await loadScript( script );
  }
  initializeThem();
  callback();
} catch ( err ) {
  reportError( err );
}
```

Reply:
I don't know for sure either, but from looking at the URLs it doesn't seem likely to me which is why I went for the parallel option.  

I'll see if I can find someone that knows the answer

Reply:
it also would bail on the remaining scripts if any one of them in the series failed.

I'm pretty sure these don't have any dependencies and the old behavior is likely broken.

Reply:
@jsnmoon @MicBosi @travisw @mattm: do any of you know whether or not the ad tracking scripts were meant to be loaded sequentially and whether or not loading the in parallel is okay?

Reply:
I'm not sure on this. @MicBosi would probably know the most here.

Reply:
@samouri @jsnajdr Hi there, yes we explicitly removed the parallel loading of the scripts in order to reduce the impact on loading times, if possible the new version should behave similarly.

Reply:
I've tested the branch using https://calypso.live/?branch=remove/dep-async&flags=ad-tracking (note the `flags=ad-tracking`) and things seem to be working but I'd like to test them again once the loading is made serial again - feel free to ping me then, also in Slack to speed things up, I'll be afk for a week starting this Friday.

Reply:
> we explicitly removed the parallel loading of the scripts in order to reduce the impact on loading times

what impact were you seeing?

Reply:
@blowery I don't recall the exact figures, but part was that we were seeing measurable differences part was staying on the safe side since those scripts even if they're marked to be loaded `async` they were still being loaded before visual elements like pictures etc. bc they are requested before, last time I checked. If you guys want to turn it back into parallel mode I haven't got strong feeling against it but you may want to double check what implications that entails. This however tends to fluctuate since we may periodically enable/disable some of the trackers. Having it serial is the safest and most user friendly choice, on the marketing side there's not need to burst-load them all at once, they also tend to generate lots of connection requests, so is good to spread them as much as possible.

Reply:
Thanks for the insights @MicBosi! I don't want to rabbit hole into too much investigation for the purposes of this PR, so I'll follow @jsnajdr's advice with a slight modification: even if one script fails it should continue to load the others.

Reply:
@MicBosi is right that if we load the ad-tracking scripts in parallel on page load, we will steal network bandwidth from more important things. Ad trackers can wait a few seconds.

When profiling initial load of `/stats`, I noticed that ad tracking scripts continue loading one by one after the initial load is finished. I liked that behavior a lot, although it's only now when I'm learning why and how it happens.

I recommend to keep loading the scripts one by one.